numba/core/cgutils.py

```diff
@@ -1013,56 +1014,59 @@ def cbranch_or_continue(builder, cond, bbtrue):
     return bbcont
 
 
-def memcpy(builder, dst, src, count):
+def memcpy(builder, dst, src, count, itemsize=1):
     """
     Emit a memcpy to the builder.
 
-    Copies each element of dst to src. Unlike the C equivalent, each element
-    can be any LLVM type.
+    Source and destination types can be any type, but the types must be the
+    same.
 
-    Assumes
-    -------
-    * dst.type == src.type
-    * count is positive
+    Call either as:
+
+    - memcpy(builder, dst, src, len), or
+    - memcpy(builder, dst, src, count, itemsize)
     """
-    # Note this does seem to be optimized as a raw memcpy() by LLVM
-    # whenever possible...
-    assert dst.type == src.type
-    with for_range(builder, count, intp=count.type) as loop:
-        out_ptr = builder.gep(dst, [loop.index])
-        in_ptr = builder.gep(src, [loop.index])
-        builder.store(builder.load(in_ptr), out_ptr)
+    if dst.type != src.type:
+        msg = f'memcpy requires the same types; got {dst.type} and {src.type}'
```

Comment:
```suggestion
        msg = f'memcpy requires the same types; got source type {src.type} and destination type {dst.type}'
```
would it be useful to specify which way around these are?